packages/shared/src/components/post/PostContentHeaderMobile.tsx

```diff
@@ -4,46 +4,27 @@ import { Button, ButtonSize, ButtonVariant } from '../buttons/Button';
 import { ArrowIcon } from '../icons';
 import { PostHeaderActions } from './PostHeaderActions';
 import { PostNavigationProps } from './common';
-import { SourceType } from '../../graphql/sources';
+import { isNullOrUndefined } from '../../lib/func';
 
 export function PostContentHeaderMobile({
   post,
   onReadArticle,
 }: Pick<PostNavigationProps, 'onReadArticle' | 'post'>): ReactElement {
   const router = useRouter();
-  const canGoBack = !!globalThis?.window?.history?.length;
-  const routedFromSquad = router?.query?.squad;
-  const squadLink = `/squads/${router.query.squad}`;
-
-  const onBackClick = () => {
-    if (canGoBack) {
-      return router.back();
-    }
-
-    if (routedFromSquad) {
-      return router.push(squadLink);
-    }
-
-    if (post.source.type === SourceType.Squad) {
-      return router.push(post.source.permalink);
-    }
-
-    return null;
-  };
+  const canGoBack =
+    globalThis?.history &&
+    !!globalThis.history.length &&
```

Comment:
Why didn't u reuse `!!globalThis?.window?.history?.length` for these?

Reply:
The condition `isNullOrUndefined(globalThis.history.state.options.shallow)` would be true if the `globalThis?.history?.state` is used.

Reply:
Changed the condition to make it easier to understand.